### PR TITLE
Allow Delete / Destroy Action

### PR DESCRIPTION
**Pull request by zrosenbauer**
I added in a the ability to destroy/delete. Please provide feedback and let me know if you need me to tweak it / change API etc. I'm actively using the version I built but felt like maybe others needed it (I'm using Cloud Run to handle preview environments for development).

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -18,6 +18,11 @@ description: |-
   Use this action to deploy a container in the Google Container Registry to Cloud Run.
 
 inputs:
+  action:
+    description: |-
+      Action to take upon the Cloud Run service, either to "deploy" or "delete".
+    required: true
```
Comment (zrosenbauer) [inline action.yml:24]: this is a breaking change FYI

File: src/cloudRun.ts [hunk 2]
```diff
@@ -311,7 +311,7 @@ export class CloudRun {
    */
   async pollService(serviceResponse: run_v1.Schema$Service): Promise<string> {
     let url = getUrl(serviceResponse);
-    const maxAttempts = 60; // Timeout after 300 seconds
+    const maxAttempts = 120; // Timeout after 300 seconds
```
Comment (zrosenbauer) [inline src/cloudRun.ts:314]: I found during deletion I needed more time BUT I'm not sure if that was a fluke... I can bump down or update the seconds*

File: src/index.ts [hunk 3]
```diff
@@ -31,16 +34,26 @@ async function run(): Promise<void> {
     const yaml = core.getInput('metadata');
     const credentials = core.getInput('credentials');
     const projectId = core.getInput('project_id');
+    const action = core.getInput('action') || 'deploy';
```
Comment (zrosenbauer) [inline src/index.ts:37]: If I remove the action.yaml requirement this will already work...

File: src/service.ts [hunk 4]
```diff
@@ -65,7 +66,10 @@ export class Service {
   readonly name: string;
 
   constructor(opts: ServiceOptions) {
-    if ((!opts.name || !opts.image) && !opts.yaml) {
+    if (
+      (!opts.name || (opts.action === 'deploy' && !opts.image)) &&
+      !opts.yaml
```
Comment (zrosenbauer) [inline src/service.ts:71]: Allow for delete w/o image

**Comment by zrosenbauer** [comment]
I know tests are failing but I'd like feedback from the maintainers that this is desired before I make any effort to resolve them...

**Comment by averikitsch** [comment]
Hi @zrosenbauer let me chat with the other maintainers about this pattern. This does seem like a usual pattern for those using the action for testing.

**Comment by zrosenbauer** [comment]
Any update here?

**Comment by averikitsch** [comment]
We would support this feature; however I am hesitant to accept this PR since we are most likely updating it to use `gcloud` instead.

**Comment by zrosenbauer** [comment]
@averikitsch yea I was a bit confused why we weren't proxying to the gcloud API. I can reopen my feature once #42 is merged.

**Pull request closed without merging**